pilot/pkg/autoregistration/controller.go

```diff
@@ -229,17 +229,24 @@ func (c *Controller) RegisterWorkload(proxy *model.Proxy, conTime time.Time) err
 }
 
 func (c *Controller) registerWorkload(entryName string, proxy *model.Proxy, conTime time.Time) error {
+	groupCfg := c.store.Get(gvk.WorkloadGroup, proxy.Metadata.AutoRegisterGroup, proxy.Metadata.Namespace)
```

Comment:
This change might not be right.. this means if a proxy reconnects but the WorkloadGroup has gone away then we will not update timestamps and the entry will expire.

Reply:
if the WorkloadGroup has gone away, we *want* the entry to expire, right? maybe i'm misunderstanding

Reply:
Probably. But that's new behavior so I want to be careful.

Reply:
There's also the OwnerRef that would cleanup entries